File: VB/E372.Module.Win/WinCustomizeTabControlViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.XtraLayout
Imports DevExpress.ExpressApp.Win.Layout

Namespace E372.Module.Web
	Public Class WinCustomizeTabControlViewController
		Inherits ViewController(Of DetailView)
		Private tabbedGroup As TabbedControlGroup
		Private layoutManager As WinLayoutManager
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			layoutManager = CType(View.LayoutManager, WinLayoutManager)
			AddHandler layoutManager.ItemCreated, AddressOf OnItemCreated
			AddHandler layoutManager.LayoutCreated, AddressOf OnLayoutCreated
		End Sub
		Private Sub OnItemCreated(ByVal sender As Object, ByVal e As ItemCreatedEventArgs)
			' Check this Id in the E372.Module/Model.DesignedDiffs.xafml file
			If e.ModelLayoutElement.Id = "TabbedGroup" Then
				tabbedGroup = CType(e.Item, TabbedControlGroup)
			End If
		End Sub
		Private Sub OnLayoutCreated(ByVal sender As Object, ByVal e As EventArgs)
			If tabbedGroup IsNot Nothing Then
				tabbedGroup.SelectedTabPageIndex = 1
			End If
		End Sub

		Protected Overrides Sub OnDeactivated()
			If layoutManager IsNot Nothing Then
				RemoveHandler layoutManager.ItemCreated, AddressOf OnItemCreated
				RemoveHandler layoutManager.LayoutCreated, AddressOf OnLayoutCreated
				layoutManager = Nothing
			End If
			tabbedGroup = Nothing
			MyBase.OnDeactivated()
		End Sub
	End Class
End Namespace

File: CS/EFCore/AccessLayoutEF/AccessLayoutEF.Win/Controllers/WinCustomizeTabControlViewController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.XtraLayout;
using DevExpress.ExpressApp.Win.Layout;

namespace AccessLayout.Win {
    public class WinCustomizeTabControlViewController : ViewController<DetailView> {
        TabbedControlGroup tabbedGroup;
        WinLayoutManager layoutManager;
        protected override void OnActivated() {
            base.OnActivated();
            layoutManager = (WinLayoutManager)View.LayoutManager;
            layoutManager.ItemCreated += OnItemCreated;
            layoutManager.LayoutCreated += OnLayoutCreated;
        }
        void OnItemCreated(object sender, ItemCreatedEventArgs e) {
            // Check this Id in the AccessLayoutEF.Module/Model.DesignedDiffs.xafml file
            if (e.ModelLayoutElement.Id == "MyTabbedGroup") {
                tabbedGroup = (TabbedControlGroup)e.Item;
            }
        }
        private void OnLayoutCreated(object sender, EventArgs e) {
            if (tabbedGroup != null) {
               tabbedGroup.SelectedTabPageIndex = 1;
            }
        }

        protected override void OnDeactivated() {
            if (layoutManager != null) {
                layoutManager.ItemCreated -= OnItemCreated;
                layoutManager.LayoutCreated -= OnLayoutCreated;
                layoutManager = null;
            }
            tabbedGroup = null;
            base.OnDeactivated();
        }
    }
}
